spring-jdbc/src/main/java/org/springframework/jdbc/core/namedparam/NamedParameterJdbcTemplate.java

```diff
@@ -429,9 +429,17 @@ protected ParsedSql getParsedSql(String sql) {
 		if (getCacheLimit() <= 0) {
 			return NamedParameterUtils.parseSqlStatement(sql);
 		}
-		synchronized (this.parsedSqlCache) {
-			return this.parsedSqlCache.computeIfAbsent(sql, NamedParameterUtils::parseSqlStatement);
+		ParsedSql parsedSql = this.parsedSqlCache.get(sql);
+		if (parsedSql == null) {
+			synchronized (this.parsedSqlCache) {
+				parsedSql = this.parsedSqlCache.get(sql);
```

Comment:
I think you can change the code more clearly with the computeIfAbsent() method.

I'd recommend you refer to [this](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-).